tests/foreman/cli/test_contentview.py

```diff
@@ -2131,3 +2136,372 @@ def test_positive_user_with_all_cv_permissions(self):
+            u'to-lifecycle-environment-id': lc_env['id'],
+        })
+        cv = ContentView.info({u'id': cv['id']})
+        environment = {'id': lc_env['id'], 'name': lc_env['name']}
+        self.assertIn(environment, cv['lifecycle-environments'])
+
+    @tier2
+    def test_positive_publish_promote_with_custom_ostree_and_other(self):
+        """Publish & Promote a content view with custom ostree and other contents
+
+        @id: 35668fa6-0a24-43ae-b562-26c5ac77e94d
+
+        @Assert: Content-view with custom ostree and other contents promoted
+        successfully
+
+        @CaseLevel: Integration
+        """
+        cv = make_content_view({u'organization-id': self.org['id']})
+        # Associate repo to CV with names.
+        repos = [
+            self.ostree_repo,
+            self.yum_repo,
+            self.docker_repo,
+        ]
+        for repo in repos:
+            ContentView.add_repository({
+                u'name': cv['name'],
+                u'organization': self.org['name'],
+                u'product': self.product['name'],
+                u'repository': repo['name'],
+            })
+        cv = ContentView.info({u'id': cv['id']})
+        self.assertEqual(
+            cv['ostree-repositories'][0]['name'],
+            self.ostree_repo['name'],
+            'Ostree Repo was not associated to CV',
+        )
+        self.assertEqual(
+            cv['yum-repositories'][0]['name'],
+            self.yum_repo['name'],
+            'Yum Repo was not associated to CV',
+        )
+        self.assertEqual(
+            cv['docker-repositories'][0]['name'],
+            self.docker_repo['name'],
+            'Docker Repo was not associated to CV',
+        )
+        # Fetch puppet module
+        puppet_result = PuppetModule.list({
+            u'repository-id': self.puppet_repo['id'],
+            u'per-page': False,
+        })
+        for puppet_module in puppet_result:
+            # Associate puppet module to CV
+            ContentView.puppet_module_add({
+                u'content-view-id': cv['id'],
+                u'uuid': puppet_module['uuid'],
+            })
+        ContentView.publish({u'id': cv['id']})
+        cv = ContentView.info({u'id': cv['id']})
+        lc_env = make_lifecycle_environment({
+            u'organization-id': self.org['id'],
+        })
+        # Promote the Published version of CV to the next env
+        ContentView.version_promote({
+            u'id': cv['versions'][0]['id'],
+            u'to-lifecycle-environment-id': lc_env['id'],
+        })
+        cv = ContentView.info({u'id': cv['id']})
+        environment = {
+            'id': lc_env['id'],
+            'name': lc_env['name'],
+        }
+        self.assertIn(environment, cv['lifecycle-environments'])
+
+
+class ContentViewRedHatOstreeContent(CLITestCase):
+    """Tests for publishing and promoting cv with RH ostree contents."""
+
+    @classmethod
+    @run_in_one_thread
```

Comment:
nitpick: We usually mark a test class instead of `setUpClass()` with `run_in_one_thread`, can't even find any example of `setUpClass` marked. Also this `setUpClass()` is already overloaded with different decorators :) so how about moving decorator to the class?